File: 2015/day20.cpp
#include <iostream>

int main(int argc, char **argv)
{
    int house = 826560;
    int highest = 0;

    while (true) {
        house += 1;
        int count = 0;

        for (int i = 1 ; i <= (int)(house / 2)+1 ; i++) {
            if (house % i == 0 && i*50 >= house) {
                count += i*11;
            }
        }
        count += house * 10;

        if (count > highest) {
            std::cout << house << " " << count << std::endl;
            highest = count;
        }

        if (count > 34000000) {
            std::cout << house << std::endl;
            return 0;
        }
    }
}

File: 2015/day20.py
import sys
import itertools


def run1(input):
    input = int(input[0].strip())
    house = 0
    highest = 0
    while True:
        house += 1
        count = 0
        for i in range(1, house+1):
            if house % i == 0:
                count += i * 10
        if count > highest:
            print(count)
            highest = count
        if count >= input:
            return house
    

if __name__ == "__main__":
    with open(sys.argv[0][:-3] + ".txt", "r") as f:
        inp = f.readlines()
        print(run1(inp))